File: lalmetaio/swig/python/check_lalmetaioswig.py
# Check SWIG Python module wrapping lalmetaio
# Author: Karl Wette, 2011, 2012

# check module load
import lal
import lalmetaio
from lalmetaio import cvar as lalmetaiocvar
from lal import cvar as lalcvar
lalcvar.lalDebugLevel = 1
print("passed module load")

# check object parent tracking
a = lalmetaio.lalmetaioswig_test_parent_map_struct()
for i in range(0, 7):
    b = a.s
    c = lalmetaiocvar.lalmetaioswig_test_parent_map.s
    lalmetaiocvar.lalmetaioswig_test_parent_map.s = lalcvar.lalswig_test_struct_const
del a, b, c
lal.CheckMemoryLeaks()
print("passed object parent tracking")

# passed all tests!
print("PASSED all tests")

File: lalmetaio/swig/octave/check_lalmetaioswig.m
## Check SWIG Octave module wrapping lalmetaio
## Author: Karl Wette, 2011, 2012

## check module load
disp("checking module load ...");
lalmetaio;
assert(exist("lalmetaio", "var"));
assert(exist("lalmetaiocvar", "var"));
lal;
assert(exist("lal", "var"));
assert(exist("lalcvar", "var"));
disp("PASSED module load");

## check object parent tracking
disp("checking object parent tracking ...");
a = lalmetaio.new_lalmetaioswig_test_parent_map_struct();
for i = 1:7
  b = a.s;
  c = lalmetaiocvar.lalmetaioswig_test_parent_map.s;
  lalmetaiocvar.lalmetaioswig_test_parent_map.s = lalcvar.lalswig_test_struct_const;
endfor
clear ans a b c;
CheckMemoryLeaks();
disp("PASSED object parent tracking");

## passed all tests!
disp("PASSED all tests");
